File: app/src/main/java/co/tpcreative/supersafe/common/presenter/BaseView.java
package co.tpcreative.supersafe.common.presenter;

import android.app.Activity;
import android.content.Context;

import java.util.List;

import co.tpcreative.supersafe.model.EnumStatus;

public interface BaseView <T> {
    void onStartLoading(EnumStatus status);
    void onStopLoading(EnumStatus status);
    void onError(String message, EnumStatus status);
    void onError(String message);
    void onSuccessful(String message);
    void onSuccessful(String message,EnumStatus status);
    void onSuccessful(String message,EnumStatus status,T object);
    void onSuccessful(String message,EnumStatus status,List<T> list);
    Context getContext();
    Activity getActivity();
}

File: app/src/main/java/co/tpcreative/supersafe/common/presenter/BaseView.kt
package co.tpcreative.supersafe.common.presenter
import android.app.Activity
import android.content.Context
import co.tpcreative.supersafe.model.EnumStatus

interface BaseView<T> {
     fun onStartLoading(status: EnumStatus)
     fun onStopLoading(status: EnumStatus)
     fun onError(message: String?)
     fun onError(message: String?, status: EnumStatus?)
     fun onSuccessful(message: String?)
     fun onSuccessful(message: String?, status: EnumStatus?)
     fun onSuccessful(message: String?, status: EnumStatus?, `object`: T?)
     fun onSuccessful(message: String?, status: EnumStatus?, list: MutableList<T>?)
     fun getContext(): Context?
     fun getActivity(): Activity?
}